cloud-sql/mysql/servlet/src/test/java/com/TestIndexServlet.java

```diff
@@ -0,0 +1,116 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.cloudsql;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.DataSource;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+
+public class TestIndexServlet {
+
+  private static List<String> requiredEnvVars =
+      Arrays.asList("MYSQL_USER", "MYSQL_PASS", "MYSQL_DB", "MYSQL_CONNECTION_NAME");
+
+  private static DataSource pool;
+
+  public static void checkEnvVars() {
+    // Check that required env vars are set
+    requiredEnvVars.forEach((varName) -> {
+      assertWithMessage(
+          String.format("Environment variable '%s' must be set to perform these tests.", varName))
+          .that(System.getenv(varName)).isNotEmpty();
+    });
+  }
+
+  private static void createTable(DataSource pool) throws SQLException {
+    // Safely attempt to create the table schema.
+    try (Connection conn = pool.getConnection()) {
+      String stmt =
+          "CREATE TABLE IF NOT EXISTS votes ( "
+              + "vote_id SERIAL NOT NULL, time_cast timestamp NOT NULL, candidate CHAR(6) NOT NULL,"
+              + " PRIMARY KEY (vote_id) );";
+      try (PreparedStatement createTableStatement = conn.prepareStatement(stmt);) {
+        createTableStatement.execute();
+      }
+    }
+  }
+
+
+  @BeforeClass
+  public static void createPool() throws SQLException {
+    checkEnvVars();
+    HikariConfig config = new HikariConfig();
+
+    config.setJdbcUrl(String.format("jdbc:mysql:///%s", System.getenv("MYSQL_DB")));
+    config.setUsername(System.getenv("MYSQL_USER")); // e.g. "root", "mysql"
+    config.setPassword(System.getenv("MYSQL_PASS")); // e.g. "my-password"
+    config.addDataSourceProperty("socketFactory", "com.google.cloud.sql.mysql.SocketFactory");
+    config.addDataSourceProperty("cloudSqlInstance", System.getenv("MYSQL_CONNECTION_NAME"));
+
+    pool = new HikariDataSource(config);
+    createTable(pool);
```

Comment:
Does this create a table that needs be to cleaned up after tests?

Reply:
added a cleanup function